src/include/sof/lib/ams.h

```diff
@@ -0,0 +1,340 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ *
+ * Copyright (c) 2022 Intel Corporation
+ *
+ * Author: Krzysztof Frydryk <krzysztofx.frydryk@intel.com>
+ *
+ */
+
+#ifndef AM_SERVICE_H
+#define AM_SERVICE_H
+
+#include <errno.h>
+#include <sof/schedule/task.h>
+#include <ipc/topology.h>
+#include <rtos/alloc.h>
+#include <sof/coherent.h>
+#include <sof/lib/uuid.h>
+
+#define AMS_INVALID_MSG_TYPE 0
+#define AMS_SERVICE_UUID_TABLE_SIZE 49
+#define AMS_INVALID_SLOT 0xFF
+#define AMS_ANY 0xFFFF
+
+#define AMS_ROUTING_TABLE_SIZE 41
```

Comment:
42 is better than 41.

That's not a good design I am afraid if any number can be picked. Please fix to something meaningful.